src/components/CreateManifestForm/CreateManifestForm.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useState, FC } from 'react';
+import {
+  Button,
+  ActionGroup,
+  Form,
+  FormGroup,
+  TextInput,
+  Title,
+  Popover,
+  FormSelect,
+  FormSelectOption
+} from '@patternfly/react-core';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon';
+import CreateManifestFormError from './CreateManifestFormError';
+import CreateManifestFormLoading from './CreateManifestFormLoading';
+import CreateManifestFormSuccess from './CreateManifestFormSuccess';
+import { SatelliteVersion } from '../../hooks/useSatelliteVersions';
+
+interface CreateManifestFormProps {
+  satelliteVersions: SatelliteVersion[];
+  handleModalToggle: () => void;
+  submitForm: (name: string, version: string) => void;
+  isLoading: boolean;
+  isError: boolean;
+  isSuccess: boolean;
+}
+
+const CreateManifestForm: FC<CreateManifestFormProps> = (props) => {
+  const { satelliteVersions, handleModalToggle, submitForm, isLoading, isError, isSuccess } = props;
+
+  const [name, setName] = useState('');
+  const [version, setVersion] = useState('Select type');
+  const [nameValidated, setNameValidated] = useState('noval');
+  const [nameHasInvalidCharacters, setNameHasInvalidCharacters] = useState(false);
+  const [versionValidated, setVersionValidated] = useState('noval');
+
+  const clearErrors = () => {
+    setNameValidated('noval');
+    setVersionValidated('noval');
+    setNameHasInvalidCharacters(false);
+  };
+
+  const handleNameChange = (value: string) => {
+    if (name.length) {
+      setNameValidated('noval');
+    }
+    setName(value);
+  };
+
+  const handleSelectChange = (value: string) => {
+    setVersionValidated('noval');
+    setVersion(value);
+  };
+
+  const validateManifestName = (name: string) => {
+    if (name.length >= 100) {
+      return false;
+    }
+    if (name.length === 0) {
+      return true;
+    }
+
+    var regExp = new RegExp('^[0-9A-Za-z_.-]+$');
+    const result = regExp.test(name);
+    return regExp.test(name);
+  };
+
+  const validateForm = () => {
```

Comment:
Is there a good external library we should be using for validation in react?  This seems kind of clunky having to do it all from scratch.  What are other cloud dot apps using?

Reply:
TLDR I think the best thing may be to move these validation methods to a separate utils / helper file and import them.  It would make them more testable and reusable in the future, and would clean this up some.

Most repos I looked at seem to be doing it "from scratch".  Several are doing it via parsing the API response when there is an error, e.g., [compliance-frontend](https://github.com/RedHatInsights/compliance-frontend/blob/master/src/SmartComponents/CreatePolicy/validate.js) or [insights-frontend-advisor](https://github.com/RedHatInsights/insights-advisor-frontend/blob/adf94ec35feade43df7643b041f57778eeafbfba/src/PresentationalComponents/Modals/AddEditTopic.js#L52)

The ones that are doing it on the front-end seem to be doing it "from scratch" but having the validation in a separate file, like [this one from cost-management](https://github.com/project-koku/koku-ui/blob/master/src/pages/costModels/components/rateForm/utils.tsx), for ex.

A couple are using Formik, which is the library currently recommended by React in their docs for forms.  It cuts down on a little of the boilerplate state management, but I’m not sure how much it will cut down in our case.  We still have to write the validation methods and still have to pass them to Patternfly.  Here is [an example](https://github.com/project-xavier/xavier-ui/blob/0d3e6ed385c4607ecd9991f4dfe5d47e1ea5937e/src/pages/ReportsUpload/ReportsUpload.tsx#L84) -  that’s also using Yup.  

[Here](https://jasonwatmore.com/post/2019/04/10/react-formik-form-validation-example) is another example from a blog post with various validation scenarios included.  

I'm inclined to just keep this but bring them to a separate helper file and import them.  I'm not sure that adding the extra libraries + code the libraries required and then plugging those into Patternfly is worth it here, but LMK what you think.